src/main/scala/io/delta/DeltaMergeBuilder.scala

```diff
@@ -132,31 +379,107 @@ case class DeltaMergeMatchedActionBuilder private[delta](
   }
 }
 
-case class DeltaMergeNotMatchedActionBuilder private[delta](
+object DeltaMergeMatchedActionBuilder {
+  /**
+   * :: Unstable ::
+   *
+   * Private method for internal usage only. Do not call this directly.
+   */
+  @Unstable
+  private[delta] def apply(
+      mergeBuilder: DeltaMergeBuilder,
+      matchCondition: Option[Column]): DeltaMergeMatchedActionBuilder = {
+    new DeltaMergeMatchedActionBuilder(mergeBuilder, matchCondition)
+  }
+}
+
+
+/**
+ * :: Evolving ::
+ *
+ * Builder class to specify the actions to perform when a source row has not matched any target
+ * Delta table row based on the merge condition, but has matched any additional condition
+ * if specified.
```

Comment:
Builder class to specify the actions to perform when a target row does not match any source row based on the merge condition but matches the `notMatchedCondition` when specified.